mono/metadata/object.c

```diff
@@ -214,7 +214,8 @@ mono_thread_set_main (MonoThread *thread)
 	static gboolean registered = FALSE;
 
 	if (!registered) {
-		MONO_GC_REGISTER_ROOT_SINGLE (main_thread, MONO_ROOT_SOURCE_THREADING, "main thread object");
+		void *key = thread->internal_thread ? MONO_UINT_TO_NATIVE_THREAD_ID (thread->internal_thread->tid) : NULL;
```

Comment:
In what situations do we expect `thread->internal_thread` to not be set here? I'm not too familiar with this code.